ddl/get.lua

```diff
@@ -52,12 +52,31 @@ local function _get_index(box_space, box_index)
     return ddl_index
 end
 
-local function get_space_sharding_key(space_name)
-    if box.space._ddl_sharding_key == nil then
+local function get_metadata(space_name, metadata_name)
+    local metadata_space_name = string.format("_ddl_%s", metadata_name)
+
+    if box.space[metadata_space_name] == nil then
+        return nil
+    end
+
+    return box.space[metadata_space_name]:get{space_name}
+end
+
+local function get_sharding_func(space_name)
+    local record = get_metadata(space_name, "sharding_func")
+    if not record then
         return nil
     end
 
-    local record = box.space._ddl_sharding_key:get{space_name}
+    if record.sharding_func_body then
+        return {body = record.sharding_func_body}
+    end
+
+    return record.sharding_func_name
```

Comment:
I would prefer explicit `~= nil` checks (for both fields), because accessing a nullable field in a middle of a tuple returns `box.NULL`. If we'll extend the space with more fields, this function may return `box.NULL` instead of `nil` and break some existing external code (if it doesn't check for `~= nil` as well).

Reply:
Updated:

```lua
local function get_sharding_func(space_name)
    local record = get_metadata(space_name, "sharding_func")
    if not record then
        return nil
    end

    if record.sharding_func_body ~= nil then
        return {body = record.sharding_func_body}
    end

    return record.sharding_func_name ~= nil and record.sharding_func_name
end
```